persistent/tests/test_persistence.py

```diff
@@ -1968,15 +2032,27 @@ def update_object_size_estimation(self, oid, new_size):
 
         return _Cache(jar)
 
-    def _makeRealCache(self, jar):
+    def _getRealCacheClass(self):
         from persistent.picklecache import PickleCachePy as PickleCache
+        return PickleCache
+
+    def _makeRealCache(self, jar):
+        PickleCache = self._getRealCacheClass()
         return PickleCache(jar, 10)
 
     def _checkMRU(self, jar, value):
-        self.assertEqual(list(jar._cache._mru), value)
+        try:
+            self.assertEqual(list(jar._cache._mru), value)
+        except AttributeError:
+            # We can't do this for the real cache.
+            assert isinstance(jar._cache, self._getRealCacheClass())
```

Comment:
Wouldn't this be clearer?

```suggestion
        if not isinstance(jar._cache, self._getRealCacheClass()):
            # We can't do this for the real cache.
            self.assertEqual(list(jar._cache._mru), value)
```

Reply:
I like the `try/except` version because it makes two assertions: the mock object works one way, and if that fails, then we must be using the real object. The `if` version kind of implicitly puts that second assertion in the condition. It seemed to me that the former is "stronger" and more likely to break if something unexpectedly changes.

But that's really all just a gut feeling. I don't mind changing it.